plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1751,6 +1763,144 @@ public void testOptimizeUsingForcedPartitioning()
+        assertUpdate("INSERT INTO " + tableName + " VALUES(1, 'INDIA', true)", 1);
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+        // Wait for TEST_METADATA_CACHE_TTL_SECONDS + 1 seconds to get the metastore cache invalidated
+        SECONDS.sleep(TEST_METADATA_CACHE_TTL_SECONDS + 1);
+        assertQueryFails("SELECT * FROM " + tableName, ".*Table '(.*)' does not exist.*");
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        String showCreateTableNew = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithComments()
+    {
+        String tableName = "test_register_table_with_comments_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, b VARCHAR, c BOOLEAN)");
+        assertUpdate("COMMENT ON TABLE " + tableName + " IS 'my-table-comment'");
+        assertUpdate("COMMENT ON COLUMN " + tableName + ".a IS 'a-comment'");
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+
+        assertThat(getTableComment(tableName)).isEqualTo("my-table-comment");
+        assertThat(getColumnComment(tableName, "a")).isEqualTo("a-comment");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithReInsert()
+    {
+        String tableName = "test_register_table_with_re_insert_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, b VARCHAR, c BOOLEAN)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(1, 'INDIA', true)", 1);
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(2, 'POLAND', false)", 1);
+
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true), (2, 'POLAND', false)");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithDroppedTable()
```

Comment:
There is nothing special about "dropped tables", since DROP TABLE wipes out the storage (is it covered with tests?)

So this is actually "testRegisterTableWithInvalidLocation" or something.

Can you simplify & rename?

Reply:
Simplified and Renamed to `testRegisterTableWithNonExistingTableLocation`